packages/editor-ui/src/components/DuplicateWorkflowDialog.vue

```diff
@@ -107,6 +109,13 @@ export default mixins(showMessage, workflowHelpers).extend({
 				return;
 			}
 
+			const allNodes = this.$store.getters.allNodes;
```

Comment:
seems you are applying the changes before duplication happens, so you are editing the original workflow. if duplication fails, these changes will apply to the current workflow. 

You should move these changes into `saveAsNewWorkflow` and updating we are sending to the save endpoint